handlers/handlers.go

```diff
@@ -150,62 +119,197 @@ func (d *CatalystAPIHandlersCollection) processUploadVOD(streamName, sourceURL,
 	return nil
 }
 
-func randomStreamName(prefix string) string {
+func randomTrailer() string {
 	const charset = "abcdefghijklmnopqrstuvwxyz0123456789"
 	const length = 8
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-
 	res := make([]byte, length)
 	for i := 0; i < length; i++ {
 		res[i] = charset[r.Intn(length)]
 	}
-	return fmt.Sprintf("%s%s", prefix, string(res))
+	return string(res)
+}
+func randomStreamName(prefix string) string {
+	return fmt.Sprintf("%s%s", prefix, randomTrailer())
 }
 
 type MistCallbackHandlersCollection struct {
 	MistClient  MistAPIClient
-	StreamCache map[string]StreamInfo
+	StreamCache *StreamCache
 }
 
-func (d *MistCallbackHandlersCollection) Trigger() httprouter.Handle {
-	return func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-		if t := req.Header.Get("X-Trigger"); t != "PUSH_END" {
-			errors.WriteHTTPBadRequest(w, "Unsupported X-Trigger", fmt.Errorf("unknown trigger '%s'", t))
-			return
+// This trigger is stream-specific and must be blocking.
+// The payload for this trigger is multiple lines, each separated by a single newline character
+// (without an ending newline), containing data as such: `stream name`\n`push target URI`
```

Comment:
I think it's hard to grasp where in the whole transcoding process the `LIVE_TRACK_LIST` trigger is called. Could you add this information to the `TranscodeSegment` function comment. I think we need a place which described the whole of process of interactions between `Catalyst API <> MistLivepeeerProc <> Mist`.